pss/pss_test.go

```diff
@@ -48,28 +50,95 @@ func TestTrojanChunkRetrieval(t *testing.T) {
 	payload := []byte("RECOVERY CHUNK")
 	topic := trojan.NewTopic("RECOVERY")
 
-	var ch chunk.Chunk
+	var monitor *Monitor
 
 	// call Send to store trojan chunk in localstore
-	if ch, err = pss.Send(ctx, testTargets, topic, payload); err != nil {
+	if monitor, err = pss.Send(ctx, testTargets, topic, payload); err != nil {
 		t.Fatal(err)
 	}
 
 	// verify store, that trojan chunk has been stored correctly
 	var storedChunk chunk.Chunk
-	if storedChunk, err = localStore.Get(ctx, chunk.ModeGetRequest, ch.Address()); err != nil {
+	if storedChunk, err = localStore.Get(ctx, chunk.ModeGetRequest, monitor.chunk.Address()); err != nil {
 		t.Fatal(err)
 	}
+	storedChunk = storedChunk.WithTagID(monitor.chunk.TagID())
 
-	if !reflect.DeepEqual(ch, storedChunk) {
+	if !reflect.DeepEqual(monitor.chunk, storedChunk) {
 		t.Fatalf("store chunk does not match sent chunk")
 	}
 
 	// check if pinning makes a difference
 
 }
 
-func newMockLocalStore(t *testing.T) *localstore.DB {
+// TestPssMonitor creates a trojan chunk
+// mocks the localstore
+// calls pss.Send method
+// updates the tag state (Stored/Sent/Synced)
+// waits for the monitor to notify the changed state
+func TestPssMonitor(t *testing.T) {
+	var err error
+	ctx := context.TODO()
+	tags := chunk.NewTags()
+
+	localStore := newMockLocalStore(t, tags)
+
+	testTargets := [][]byte{
+		{57, 120},
+		{209, 156},
+		{156, 38},
+		{89, 19},
+		{22, 129}}
+	payload := []byte("RECOVERY CHUNK")
+	topic := trojan.NewTopic("RECOVERY")
+
+	var monitor *Monitor
+
+	pss := NewPss(localStore, tags)
+
+	// call Send to store trojan chunk in localstore
+	if monitor, err = pss.Send(ctx, testTargets, topic, payload); err != nil {
+		t.Fatal(err)
+	}
+
+	// increment stored on the tag, expect state stored on the channel
+	monitor.tag.Inc(chunk.StateStored)
```

Comment:
do it in a loop ranging over the 3 states, like you do in the code, since you  write the statechangevalid once, no need for a separate function either, a little more elegant perhaps

Reply:
I refactored it, this is much better.